src/cio_websocket.c

```diff
@@ -166,22 +180,18 @@ static bool is_status_code_invalid(uint16_t status_code)
 	return true;
 }
 
-static void close_frame_written(struct cio_buffered_stream *bs, void *handler_context, const struct cio_write_buffer *buffer, enum cio_error err)
+static void close_frame_written(struct cio_buffered_stream *bs, void *handler_context, enum cio_error err)
 {
 	(void)bs;
-	(void)buffer;
+	(void)handler_context;
 	(void)err;
-	struct cio_websocket *ws = (struct cio_websocket *)handler_context;
-	if (ws->ws_flags.self_initiated_close == 0) {
-		close(ws);
-	}
+	// TODO: we could emit a shutdown(WR) here.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=c%3AS1135)